### PR TITLE
Add `xun create-cryptographic-identities` command

### PR DESCRIPTION
**Pull request by JensGM**
This can be used to configure secure dask

### CODE REVIEW

File: xun/infrastructure/crypto.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        .serial_number(x509.random_serial_number())
+
+        .add_extension(
+            x509.BasicConstraints(ca=False, path_length=None),
+            critical=True)
+
+        .not_valid_before(now)
+        .not_valid_after(now + timedelta(days=60))
+
+        .sign(ca.key, hashes.SHA256(), default_backend())
```
Comment (vero-so) [inline xun/infrastructure/crypto.py:107]: Can you check how you are setting things up against this tutorial? https://cryptography.io/en/latest/x509/tutorial/

**Review by vero-so — COMMENTED** [review]
Check out the other setup of the certificate before merging.
The help messages are less relevant, but could be nice to have it consistent in an imperative form starting with a capital?

File: xun/tests/test_crypto.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    assert tmpdir.join('xun-dask-client-private-key.pem').check()
+    assert tmpdir.join('xun-dask-client-public-key.pem').check()
+    assert tmpdir.join('xun-dask-client-cert.pem').check()
+    assert tmpdir.join('xun-dask-scheduler-private-key.pem').check()
+    assert tmpdir.join('xun-dask-scheduler-public-key.pem').check()
+    assert tmpdir.join('xun-dask-scheduler-cert.pem').check()
+    assert tmpdir.join('xun-dask-worker-private-key.pem').check()
+    assert tmpdir.join('xun-dask-worker-public-key.pem').check()
+    assert tmpdir.join('xun-dask-worker-cert.pem').check()
+    assert tmpdir.join('xun-tls.yml').check
```
Comment (JensGM) [inline xun/tests/test_crypto.py:66]: Parens

Comment (vero-so) [reply xun/tests/test_crypto.py:66]: -thesis :)

**Review by vero-so — APPROVED** [review]
Looks good!

**Pull request merged**